src/Query/FetchProductQuery.php

```diff
@@ -5,29 +5,36 @@
 namespace App\Query;
 
 use Akeneo\Pim\ApiClient\AkeneoPimClientInterface;
-use Akeneo\Pim\ApiClient\Exception\NotFoundHttpException;
+use Akeneo\Pim\ApiClient\Exception\NotFoundHttpException as AkeneoNotFoundHttpException;
 use Akeneo\Pim\ApiClient\Search\Operator;
 use Akeneo\Pim\ApiClient\Search\SearchBuilder;
 use App\PimApi\Model\Product;
 use App\PimApi\Model\ProductValue;
+use App\PimApi\PimCatalogApiClient;
 use App\PimApi\ProductValueDenormalizer;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
 /**
- * @phpstan-type RawProduct array{identifier: string, family: string|null, values: array<string, array{array{locale: string|null, scope: string|null, data: mixed}}>}
+ * @phpstan-type RawProduct array{uuid: string, family: string|null, values: array<string, array{array{locale: string|null, scope: string|null, data: mixed}}>}
  * @phpstan-type RawFamily array{code: string, attribute_as_label: string}
  */
 final class FetchProductQuery
 {
     public function __construct(
         private AkeneoPimClientInterface $pimApiClient,
         private ProductValueDenormalizer $productValueDenormalizer,
+        private readonly PimCatalogApiClient $catalogApiClient,
     ) {
     }
 
-    public function fetch(string $identifier, string $locale): Product
+    public function fetch(string $catalogId, string $identifier, string $locale): Product
     {
-        /** @var RawProduct $rawProduct */
-        $rawProduct = $this->pimApiClient->getProductApi()->get($identifier);
+        try {
+            /** @var RawProduct $rawProduct */
+            $rawProduct = $this->catalogApiClient->getProduct($catalogId, $identifier);
+        } catch (\Exception $e) {
+            throw new NotFoundHttpException('PIM API replied with a 404', $e);
```

Comment:
You should not throw `HttpKernel\Exception\NotFoundHttpException` in your query.
A query could be called from another context, like a command.
It's better to throw an business-like exception, like `Akeneo\Pim\ApiClient\Exception\NotFoundHttpException` or your own custom exception.
Then, you catch it in the caller (eg, your controller) where you deal with it accordingly.